### ISSUE TITLE
Add ability to automatically approve requests based on criteria

### ISSUE THREAD

**Issue by augustuen** [Fallenbagel/jellyseerr#1184]
### Description

When running a Jellyseerr server for other people, you may not want *all* their requests to be automatically approved. Or you may want to enforce certain settings for certain requests (like specific root folders or quality profiles for specific types of movies or series) 

For these reasons, it would be nice to be able to further restrict auto-approval, or even make the auto-approval process modify requests according to the admin's wishes.

### Desired Behavior

I'd like the ability to define rules with a number of conditions, to be automatically checked whenever a user submits a new request. If a request fulfills all the conditions of a rule, that request would be approved. 

Conditions could for example be genre, release year, number of seasons (requested), which user made the request, etc.

Possibly, the rules could also dictate certain parts of the request, such as the destination server, quality profiles, root folders, or tags. 

### Additional Context

I've already done some work, to better illustrate my vision. I did this with Overseerr, but seeing as Jellyseerr seems to be considerably more active in development, I figured it a better fit. Currently it's mostly UI, but implementing the logic seems like the logical (heh) next step to me. I've provided some screenshots (from Overseerr) below:

**List of defined approval rules:**
![Skjermbilde](https://github.com/user-attachments/assets/3fd33efe-3bd2-4229-8184-a4a802e3cc15)

**Editing an auto-approval rule:**
![Skjermbilde2](https://github.com/user-attachments/assets/2e372ccf-b2bc-4c4c-b60b-05e60f399471)

**Modifying the wanted value of a condition**
![Skjermbilde3](https://github.com/user-attachments/assets/c8f64fb0-9f67-4988-9a1a-65f2f169f165)

I'd love to continue working on this, but with my limited experience (especially with frontend), I feel I've reached a point where I would need to discuss it with others to see the path ahead. 

### Code of Conduct

- [X] I agree to follow Jellyseerr's Code of Conduct

**Comment by gauthier-th** [comment]
You could also open a PR here, and mark it as draft PR, so people can see what work have you done

**Comment by augustuen** [comment]
Yup, I can do that. I just figured I'd throw it up as an issue first to see what the interest/thoughts were. Especially since I've been doing this based on Overseerr, so it'll take a bit of time/effort to move over to Jellyseerr. I'll get on it right away.
